src/image/TransmissionCurve.cc

```diff
@@ -0,0 +1,851 @@
+    {
+        LSST_THROW_IF_NE(
+            wavelengths.getSize<0>(), throughput.getSize<0>(),
+            pex::exceptions::LengthError,
+            "Length of wavelength array (%d) does not match size of throughput array (%d)"
+        );
+        int status = ::gsl_interp_init(_interp.get(), _wavelengths.getData(), _throughput.getData(),
+                                       _throughput.getSize<0>());
+        LSST_CHECK_GSL(pex::exceptions::LogicError, status);
+    }
+
+    SampleDef getNaturalSampling() const override {
+        return SampleDef(_wavelengths.front(), _wavelengths.back(), _naturalSamplingSize);
+    }
+
+    std::pair<double,double> getThroughputAtBounds() const override {
+        return _atBounds;
+    }
+
+    void sampleAt(
+        geom::Point2D const &,
+        ndarray::Array<double const,1,1> const & wavelengths,
+        ndarray::Array<double,1,1> const & out
+    ) const override {
+        LSST_THROW_IF_NE(
+            wavelengths.getSize<0>(), out.getSize<0>(),
+            pex::exceptions::LengthError,
+            "Length of wavelength array (%d) does not match size of output array (%d)"
+        );
+        CPtr<::gsl_interp_accel> accel(::gsl_interp_accel_alloc(), &gsl_interp_accel_free);
+        auto wlIter = wavelengths.begin();
+        for (auto outIter = out.begin(); outIter != out.end(); ++outIter, ++wlIter) {
+            double & y = *outIter;
+            if (*wlIter < _wavelengths.front()) {
+                y = _atBounds.first;
+            } else if (*wlIter > _wavelengths.back()) {
+                y = _atBounds.second;
+            } else {
+                int status = ::gsl_interp_eval_e(_interp.get(), _wavelengths.getData(), _throughput.getData(),
+                                                 *wlIter, accel.get(), &y);
+                if (status) {
+                    throw LSST_EXCEPT(
+                        pex::exceptions::InvalidParameterError,
+                        (boost::format("Invalid wavelength value: %g") % (*wlIter)).str()
+                    );
+                }
+            }
+        }
+    }
+
+    bool isPersistable() const override { return true; }
+
+protected:
+
+    // transforming a ConstantTransmissionCurve is a no-op
+    std::shared_ptr<TransmissionCurve> _transformImpl(
+        std::shared_ptr<TransmissionCurve> self,
+        std::shared_ptr<geom::TransformPoint2ToPoint2> transform
+    ) const override {
+        return self;
+    }
+
+    std::string getPersistenceName() const override { return "ConstantTransmissionCurve"; }
+
+    std::string getPythonModule() const override { return "lsst.afw.image"; }
+
+    void write(OutputArchiveHandle& handle) const override;
+
+private:
+    std::pair<double,double> _atBounds;
+    ndarray::Array<double,1,1> _throughput;
+    ndarray::Array<double,1,1> _wavelengths;
+    int _naturalSamplingSize;
+    CPtr<::gsl_interp> _interp;
+};
+
+struct ConstantPersistenceHelper {
```

Comment:
I don't think there's much to differentiate in this case, since the helper class doesn't actually need access to the class's private content, and it's in an anonymous namespace regardless.  I think it's slightly more common for these persistence helpers to not be nested, but I can't think of any other arguments for or against.